backend/server-connection/src/main/java/org/sonarsource/sonarlint/core/serverconnection/storage/ServerInfoStorage.java

```diff
@@ -54,11 +54,11 @@ public Optional<StoredServerInfo> read() {
     return rwLock.read(() -> Files.exists(storageFilePath) ? Optional.of(adapt(ProtobufFileUtil.readFile(storageFilePath, Sonarlint.ServerInfo.parser()))) : Optional.empty());
   }
 
-  private static Sonarlint.ServerInfo adapt(ServerInfo serverInfo) {
-    return Sonarlint.ServerInfo.newBuilder().setVersion(serverInfo.getVersion()).build();
+  private static Sonarlint.ServerInfo adapt(ServerStatusInfo serverStatus, boolean isMQRMode) {
+    return Sonarlint.ServerInfo.newBuilder().setVersion(serverStatus.getVersion()).setIsMqrMode(isMQRMode).build();
   }
 
   private static StoredServerInfo adapt(Sonarlint.ServerInfo serverInfo) {
-    return new StoredServerInfo(Version.create(serverInfo.getVersion()));
+    return new StoredServerInfo(Version.create(serverInfo.getVersion()), serverInfo.getIsMqrMode());
```

Comment:
What happens if we try to read a storage that has no value for `isMqrMode`, if the storage has not yet been synchronized?

Reply:
The storage is per connection anyway, and the client should tell if a given connection is SQ or SC